README.md

```diff
@@ -30,48 +30,91 @@ Options:
+```mermaid
+graph LR
+  A(Client) -->|Fetch toggles| B((Edge))
+  B-->|Fetch features| C((Unleash))
+```
+
+Edge mode is the "standard" mode for Unleash Edge and the one you should default to in most cases. It connects to an upstream node, such as your Unleash instance, and uses that as the source of truth for feature toggles.
 
-If you'd rather use a simple key like `secret-123`, any query against `/api/client/features` will receive the dump passed in on the command line.
+Another than connecting Edge directly to your Unleash instance, it's also possible to connect to another Edge instance (_daisy chaining_). You can have as many Edge nodes as you'd like between the Edge node your clients are accessing and the Unleash server, and it's also possible for multiple nodes to connect to a single upstream one. Depending on your architecture and requirements this can be a powerful feature, offering you flexibility and scalability when planning your implementation.
 
-Any query against `/api/frontend` or `/api/proxy` with a valid key will receive only enabled toggles.
+```mermaid
+graph LR
+  A(Client 1) -->|Fetch toggles| C((Edge 1))
+  B(Client 2) -->|Fetch toggles| D((Edge 2))
+  C-->|Fetch features| E((Edge 3))
+  D-->|Fetch features| E
+  E-->|Fetch features| F((Unleash))
+```
+
+This means that, in order to start up, Edge mode needs to know where the upstream node is. This is done by passing the `--unleash-url` command line argument or setting the `UNLEASH_URL` environment variable.
+
+By default, Edge mode uses in-memory to store the features it fetches from the upstream node. However, you may want to use a more persistent storage solution. Edge supports Redis as well, and you can configure it by passing in the `--redis-url` command line argument or setting the `REDIS_URL` environment variable.
+
+Edge mode also supports dynamic tokens, meaning that when using a [client token](https://docs.getunleash.io/reference/api-tokens-and-client-keys#client-tokens) with the `/api/client/features` endpoint, even without previously knowing about it, Edge will validate upstream and fetch its respective features. After that, it gets added to the list of known tokens that gets periodically synced, making sure it is a valid token and its features are up-to-date.
+
+[Front-end tokens](https://docs.getunleash.io/reference/api-tokens-and-client-keys#front-end-tokens) can also be used with `/api/frontend` and `/api/proxy` endpoints, however they are not allowed to fetch features upstream. In order to use these tokens correctly and make sure they return the correct information, it's important that the features they are allowed to access are already present in that Edge node's features cache. The easiest way to ensure this is by passing in at least one client token as one of the command line arguments, ensuring it has access to the same features as the front-end token you'll be using.
```

Comment:
```suggestion
[Front-end tokens](https://docs.getunleash.io/reference/api-tokens-and-client-keys#front-end-tokens) can also be used with `/api/frontend` and `/api/proxy` endpoints, however they are not allowed to fetch features upstream. In order to use these tokens correctly and make sure they return the correct information: e.g. by passing in at least one client token as one of the command line arguments, ensuring it has access to the same features as the front-end token you'll be using.
```
I'd try to provide an example.

I also removed `it's important that the features they are allowed to access are already present in that Edge node's features cache` because I'm not sure how customers will be able to notice this. Perhaps the explanation above is enough

Reply:
Unsure about this change, I'll leave it up for discussion a bit longer if anyone else wants to chime in.